code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 
 	succeed_activate()
 	add_cooldown()
+
+// ***************************************
+// *********** Abduct
+// ***************************************
+
+/datum/action/ability/activable/xeno/abduct
+	name = "Abduct"
+	action_icon_state = "abduct"
+	action_icon = 'icons/Xeno/actions/praetorian.dmi'
+	desc = "Throw your tail out and hook in any humans caught in it. Ends prematurely if blocked or hits anything dense."
+	ability_cost = 50
+	cooldown_duration = 16 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ABDUCT,
+	)
+	/// The turf in front of the owner when the projectile was initially fired.
+	var/turf/initial_turf
+	/// A list of turfs that will be used to get mobs who will get affected by the ability.
+	var/list/turf/turf_line
+	/// The created effects that need to be deleted later.
+	var/list/obj/effect/xeno/abduct_warning/telegraphed_atoms
+	/// A timer for when to conclude the ability.
+	var/ability_timer
+
+
+/datum/action/ability/activable/xeno/abduct/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(ability_timer)
+		if(!silent)
+			target.balloon_alert(xeno_owner, "already abducting")
+		return FALSE
+	// Getting direction without calling face_atom on the owner.
+	var/direction_to = get_dir(xeno_owner, target)
+	var/dx = target.x - xeno_owner.x
+	var/dy = target.y - xeno_owner.y
+	if(!dx && !dy)
+		if(target.pixel_y > 16)
+			direction_to = NORTH
+		else if(target.pixel_y < -16)
+			direction_to = SOUTH
+		else if(target.pixel_x > 16)
+			direction_to = EAST
+		else if(target.pixel_x < -16)
+			direction_to = WEST
+	else if(abs(dx) < abs(dy))
+		if(dy > 0)
+			direction_to = NORTH
+		else
+			direction_to = SOUTH
+	else
+		if(dx > 0)
+			direction_to = EAST
+		else
+			direction_to = WEST
+
+	var/turf/initial_turf_pre = get_step(xeno_owner, direction_to)
+	var/list/turf/turf_line_pre = get_turf_line(initial_turf_pre, target, 1)
+	if(!turf_line_pre.len) // Being really nice by preventing them from using the ability if it would of done nothing.
+		if(!silent)
+			target.balloon_alert(xeno_owner, "blocked")
+		return FALSE
+
+/datum/action/ability/activable/xeno/abduct/use_ability(atom/movable/target)
```

Comment:
Don't just change arg types. target is an atom, if you need to typecast it as an AM, do so in the proc.